documentation/index.md

```diff
@@ -0,0 +1,639 @@
+
+This is handled inside the `GeocodeViewModel`. When the search text property is updated the `GetLocationSuggestionsAsync` method is called. A check is first performed to ensure the locator supports suggestions. When [creating your own locator](https://pro.arcgis.com/en/pro-app/tool-reference/geocoding/create-address-locator.htm), you can enable suggestions and thus be able to take advantage of this functionality.
+
+```csharp
+/// <summary>
+/// Gets list of suggested locations from the locator based on user input
+/// </summary>
+private async Task GetLocationSuggestionsAsync(string userInput)
+{
+    // make sure input is defined and suggestions are supported
+    if (Locator?.LocatorInfo?.SupportsSuggestions ?? false && !string.IsNullOrEmpty(userInput))
+    {
+        // restrict the search to return no more than 10 suggestions
+        // set preferred search location to location around current map center
+        var suggestParams = new SuggestParameters { MaxResults = 10, PreferredSearchLocation = AreaOfInterest?.TargetGeometry as MapPoint, };
+
+        // get suggestions for the text provided by the user
+        var suggestions = await Locator.SuggestAsync(userInput, suggestParams);
+        var s = new ObservableCollection<string>();
+        foreach (var suggestion in suggestions)
+        {
+            s.Add(suggestion.Label);
+        }
+        SuggestionsList = s;
+    }
+}
+```
+
+## Searching from a suggestion
+
+Once a suggestion in the list has been selected by the user, the suggested address is geocoded using the `LocatorTask.GeocodeAsync` function. Along with the address, specific geocoding parameters can be set to tune the results. For example, in the Maps App, we set the `PreferredSearchLocation` property to prioritize results closer to the center of the map.
+
+```csharp
+/// <summary>
+/// Get location searched by user from the locator
+/// </summary>
+private async Task<GeocodeResult> GetSearchedLocationAsync(string geocodeAddress)
+{
+    // Locate the searched feature
+    var geocodeParameters = new GeocodeParameters
+    {
+        MaxResults = 1,
+        PreferredSearchLocation = AreaOfInterest?.TargetGeometry as MapPoint,
+    };
+
+    // return the first match
+    var matches = await Locator.GeocodeAsync(geocodeAddress, geocodeParameters);
+    return matches.FirstOrDefault();
+}
+```
+
+| Android | WPF |
+|:-------:|:---:|
+|<img src="/documentation/images/geocode_and.png" width="450"/>|<img src="/documentation/images/geocode_wpf.png"/>|
+
+## Routing
+
+Getting routing directions in the Maps App is easy with the [Runtime SDK](https://developers.arcgis.com/features/directions/) and the [Esri World Routing Service](https://route.arcgis.com/arcgis/rest/services/World/Route/NAServer/Route_World). Or you can [customize](https://doc.arcgis.com/en/arcgis-online/administer/configure-services.htm#ESRI_SECTION1_567C344D5DEE444988CA2FE5193F3CAD) your routing service for your organization.
```

Comment:
[Esri World Routing Service] leads to a 403 and isn't very helpful. Can we swap for something else? I'm not sure which I like more:

- https://developers.arcgis.com/features/directions/
- https://developers.arcgis.com/rest/network/api-reference/overview-of-network-analysis-services.htm

Thoughts?

Reply:
I'm guessing the original intention in this doc was to reference the service used by the app (see below), but I've replaced it with the feature overview page so that it links to something that has useful, readable info.

https://github.com/Esri/maps-app-dotnet/blob/8f65f7e92239df7dae248c55244a16c08c1f9970/src/MapsApp.Shared/Helpers/Configuration.cs#L34